h/static/scripts/annotator/plugin/textselection.coffee

```diff
@@ -88,9 +88,8 @@ class Annotator.Plugin.TextSelection extends Annotator.Plugin
 
     for range in selectedRanges
       container = range.commonAncestor
-      # TODO: what is selection ends inside a different type of highlight?
-      if @Annotator.TextHighlight.isInstance container
-        container = @Annotator.TextHighlight.getIndependentParent container
+      if @$(container).hasClass('annotator-hl')
+        container = @$(container).parents(':not([class^=annotator-hl])')[0]
```

Comment:
If we take the latest annotator v1.2.x these two lines can be removed.
https://github.com/openannotation/annotator/commit/368b5cf4b39bda76464bab00ec6c44efed335694